sql/api/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBaseParser.g4

```diff
@@ -823,13 +823,17 @@ identifierComment
 
 relationPrimary
     : identifierReference temporalClause?
-      sample? tableAlias                                    #tableName
+      sample? optionsClause? tableAlias                        #tableName
     | LEFT_PAREN query RIGHT_PAREN sample? tableAlias       #aliasedQuery
     | LEFT_PAREN relation RIGHT_PAREN sample? tableAlias    #aliasedRelation
     | inlineTable                                           #inlineTableDefault2
     | functionTable                                         #tableValuedFunction
     ;
 
+optionsClause
+    : WITH OPTIONS options=propertyList
```

Comment:
Shall we use `WITH` instead of `WITH OPTIONS`? [postgresql syntax](https://www.postgresql.org/docs/current/sql-createtable.html) uses `WITH`

Reply:
Sounds good to me.

Reply:
Change back to WITH OPTIONS ,  checking is it @cloud-fan 's suggestion?

Reply:
Shall we make `OPTIONS` optional so that it's more flexible?

Reply:
There is a discontinuity in the comments.... why did it flip flop? I'm also in favor of just WITH.